lib/rails_multitenant/global_context_registry.rb

```diff
@@ -91,6 +91,14 @@ def as_current_id(id)
           self.current_id = old_id
         end
 
+        def as_current(model)
+          old_model = current
+          self.current = model
```

Comment:
Should you verify that the model is actually an instance of the class? I.e.

```ruby
raise "#{model} is not a #{self}" if model.present? && !model.is_a?(self)
```

Reply:
Hey Tim! Sure, I'll add that. Good idea.